Makefile

```diff
@@ -18,18 +22,60 @@ clean: ## delete most git-ignored files
 	find . -name '*.pyc' -exec rm -f {} +
 	find . -name '*.pyo' -exec rm -f {} +
 	find . -name '*~' -exec rm -f {} +
+	coverage erase
+	rm -fr *.egg-info
 
 requirements: ## install environment requirements
 	pip install -r requirements/base.txt
 
+install-automation-reqs: ## install tox
+	$(TOX) pip install -r requirements/tox.txt
+
+tox-test-reqs: install-automation-reqs
+	$(TOX) pip install -r requirements/test.txt --exists-action w
+
 upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
 upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
 	pip install -qr requirements/pip-tools.txt
 	# Make sure to compile files after any other files they include!
 	$(PIP_COMPILE) -o requirements/pip-tools.txt requirements/pip-tools.in
 	$(PIP_COMPILE) -o requirements/base.txt requirements/base.in
 	$(PIP_COMPILE) -o requirements/test.txt requirements/test.in
+	$(PIP_COMPILE) -o requirements/tox.txt requirements/tox.in
+
+quality: clean tox-test-reqs## check coding style with pycodestyle and pylint
+	$(TOX) pycodestyle ./eox_nelp
+	$(TOX) pylint ./eox_nelp --rcfile=./setup.cfg
+	$(TOX) isort --check-only --diff ./eox_nelp
+
+python-test: clean tox-test-reqs## Run test suite.
+	$(TOX) coverage run --source ./eox_nelp manage.py test
+	$(TOX) coverage report -m --fail-under=71
+
+complexity: clean tox-test-reqs## Run complexity suite with flake8.
+	$(TOX) flake8  --max-complexity 10 eox_nelp
+
+run-tests: python-test quality complexity
+
+##-----------------------------Translations section-------------------------
+extract-translations: ## extract strings to be translated, outputting .mo files
+	./manage.py makemessages -l ar   -i manage -i setup -i "venv/*"
+
+compile-translations: ## compile translation files, outputting .po files for each supported language
+	cd eox_nelp && ../manage.py compilemessages
+
+detect-changed-source-translations:
+	cd eox_nelp && i18n_tool changed
+
+pull-translations: ## pull translations from Transifex
+	tx pull -af --mode reviewed
+
+push-translations: ## push source translation files (.po) from Transifex
+	tx push -s
+
+dummy-translations: ## generate dummy translation (.po) files
```

Comment:
I have no idea the purpose of this, could you explain me  that ?

Reply:
This is if we are going to configure transifex cli in the future, this would be removed?

Reply:
the dummy translations are for testing translations based in esperanto. But actually it doesn't generate me anything. May be opened used in test.
https://github.com/openedx/edx-platform/blob/db32ff2cdf678fa8edd12c9da76a76eef0478614/pavelib/paver_tests/test_extract_and_generate.py#L72

Should I removed it?
![image](https://user-images.githubusercontent.com/51926076/217921518-73e79061-5ec8-469c-8628-647ebc9c4934.png)

Reply:
I think we won't use this, so we should remove it

Reply:
Removed in [5f93565](https://github.com/eduNEXT/eox-nelp/pull/19/commits/5f93565ef247486ad60397d087de0a0d2002f555)